### PR TITLE
Password encryption and jwt refactor

### LINKED ISSUE #51: Don't store plaintext password in db
### Describe the feature or problem you’d like to solve
use bcrypt on the passwords stored in db when users are created

### Proposed solution
storing plaintext passwords is bad

### LINKED ISSUE #52: Don't fallback on 'testing' for the JWT_SECRET, use .env
### Describe the feature or problem you’d like to solve
put JWT_SECRET in the .env and .env.sample files and load the secret from there instead of defaulting to 'testing'

### PR DESCRIPTION
**Pull request by Arturo-Lopez**
# Related Issue/Addition to code

- Fixes #51, fixes #52
 
## Type of change

- [ ] New feature (non-breaking change which adds functionality)

# Proposed Changes
- Added [bcryptjs](https://github.com/dcodeIO/bcrypt.js) to handle passwords safely
- Refactor jsonwebtokens implementation and added `JWT_SECRET` in `.env.sample`

# Additional Info
- The default value for token signing is none, JWT_SECRET is required in `.env` in order to work properly

# Screenshots
none

### CODE REVIEW

File: api/src/lib/passwordEncryption.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import bcrypt from 'bcryptjs';
+
+export const getPasswordHash = async (password: string): Promise<string> => {
+  return bcrypt.hashSync(password, 10);
```
Comment (codyseibert) [inline api/src/lib/passwordEncryption.ts:4]: I'd prefer if you'd invoke the async method if possible

File: api/src/lib/passwordEncryption.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+};
+
+export const isValidPassword = async (
+  password: string,
+  passwordHash: string,
+): Promise<boolean> => {
+  return bcrypt.compareSync(password, passwordHash);
```
Comment (codyseibert) [inline api/src/lib/passwordEncryption.ts:11]: I'd prefer if you'd invoke the async method if possible

**Comment by codyseibert** [comment]
@Arturo-Lopez thanks for helping out.  Are you on discord by chance?

**Review by codyseibert — APPROVED** [review]

**Comment by Arturo-Lopez** [comment]
> @Arturo-Lopez thanks for helping out. Are you on discord by chance?
@codyseibert yes I'm currently on discord

File: api/src/lib/passwordEncryption.ts (hunk 1 above)
Comment (Arturo-Lopez) [reply api/src/lib/passwordEncryption.ts:4]: Ok, I'll commit the change

File: api/src/lib/passwordEncryption.ts (hunk 2 above)
Comment (Arturo-Lopez) [reply api/src/lib/passwordEncryption.ts:11]: Ok, I'll commit the change

**Pull request merged**